### DISCUSSION TITLE
Welcome to langchain-chatgpt Discussions!

### DISCUSSION THREAD

**Discussion by melogabriel** [melogabriel/langchain-chatgpt#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Tamnguyen0880** [comment]
Hi,

Im looking to build an data analytics tool using natural language prompts to analyse and visualise data. Outputs can be structured data (numbers), statistical summary, editable charts (allow users to change chart types, axis name, colour, tittle, AND add filters etc.) on the chart itself. Looking at your work using Langchain & GPT, just wondering if you are interested in building up this tool for a fee? This will need to be enterprise grade (speed, scalability to handle large csv/excel/googlesheet/DB files). 
Please let me know if you are interested.

Regards,
Tam

**Reply by melogabriel** [reply]
What would this job look like? Would we have a full team to work on the tool? I ask because I don't think it's possible to release a quality tool without a complete team working.

**Reply by Tamnguyen0880** [reply]
We dont have a full team, only a group of 3 programmers specialised in data analytics and python
About the tool, objective is to allow users interact with their structured data (either in csv, excel, googlesheet, or DB), below are objectives:
1. Load files
2. Analyse data using natural languages: statistical summary,  correlation analysis between 2 or multiple parameters, 
3. Visualise in charts using natural language prompts, then make edits on the chart (change chart type, tittles, colour, size, etc.)

Would be interested in your thoughts on how big the team needs to be? and release in stages so wont need to be super refine and all functionalities at once.

**Reply by melogabriel** [reply]
So, okay. I was thinking more from a business perspective, I think that a tool is not only made with backend programmers, but also with a frontend and UX specialist.

However if the idea is to do something simple, I am interested in being part of it.

**Reply by Tamnguyen0880** [reply]
Yea, we do have some others to develop front end and back-end for the tool. The first step is to create a sound approach that will be scalable, stable, responsive (speed), and minimal token usage if using OpenAI LLM. For the tool, it is more important to be functional using something like Streamlit, and not focus on visual initial nor deploy-ability (e.g SDK or API). Below are a bit more details of the requirements. Let me know if you are interested and be able to come up with approach, estimates your efforts and costs. Other team members will use your work to package into deployable toolsets

**Reply by Tamnguyen0880** [reply]
![Data Analytics Requirements](https://github.com/melogabriel/langchain-chatgpt/assets/129024872/f8e0ac20-a15e-419e-8893-06faa90dce95)

**Reply by Tamnguyen0880** [reply]
Here is my email tam.nguyen@neuralpit.com if you would like to discuss further. We can set up a meeting in Teams 
Regards,
Tam

**Comment by Tamnguyen0880** [comment]
Starting point would be something like this https://github.com/wgryc/phasellm/tree/main/demos-and-products/researchllm

**Reply by Tamnguyen0880** [reply]
Similar tool is https://www.chatcsv.co
This tool can interpret what data in each column mean, quite accurately, can do simple analytics and simple plots
